src/core/src/energy/functions.rs

```diff
@@ -104,6 +104,69 @@ impl PairPotential for LennardJones {
     }
 }
 
+/// Weeks Chandler Andersen Potential (WCA)
+///
+/// The following energy expression is used: `V(r) = 4 *
+/// epsilon * ((sigma/r)^12 - (sigma/r)^6) + epsilon` if `r < 2^1/6 * sigma
+/// 0 if r >= 2^1/6 * sigma`
+/// where `sigma` is the WCA
+/// distance constant, and `epsilon` the WCA energetic constant.
+/// # Examples
+///
+/// ```
+/// use lumol_core::energy::Potential;
+/// use lumol_core::energy::WeeksChandlerAndersen;
+///
+/// let potential = WeeksChandlerAndersen{sigma: 2.0, epsilon: 10.0};
+/// assert_eq!(potential.energy(2.0), 10.0);
+/// assert_eq!(potential.energy(3.0), 0.0);
+/// assert_eq!(potential.energy(1.0), 161290.0);
+///
+/// assert_eq!(potential.force(2.0), 120.0);
+/// ```
+
+#[derive(Clone, Copy)]
+pub struct WeeksChandlerAndersen {
+    /// Distance constant for the WeeksChandlerAndersen Potential
+    pub sigma: f64,
+    /// Energy Constant for the WeeksChandlerAndersen Potential
+    pub epsilon: f64,
+}
+
+impl Potential for WeeksChandlerAndersen {
+    fn energy(&self, r: f64) -> f64 {
+        if r < f64::powf(2., 1./6.) * self.sigma {
+            let s6 = f64::powi(self.sigma / r, 6);
+            4.0 * self.epsilon * (s6 * s6 - s6) + self.epsilon
+        } else {
+            0.
+        }
+    }
+
+    fn force(&self, r: f64) -> f64 {
+        if r < f64::powf(2., 1./6.) * self.sigma {
+            let s6 = f64::powi(self.sigma / r, 6);
+            -24.0 * self.epsilon * (s6 - 2.0 * s6 * s6) / r
+        } else {
+            0.
+        }
+    }
+}
+
+impl PairPotential for WeeksChandlerAndersen {
+    fn tail_energy(&self, cutoff: f64) -> f64 {
+        if cutoff < f64::powf(2., 1./6.) * self.sigma {
+            panic!("Error: WCA Cutoff too small: {} < {}", cutoff, f64::powf(2., 1./6.) * self.sigma);
+        }
+        0.0
+    }
+    fn tail_virial(&self, cutoff: f64) -> f64 {
+        if cutoff < f64::powf(2., 1./6.) * self.sigma {
+            panic!("Error: WCA Cutoff too small: {} < {}", cutoff, f64::powf(2., 1./6.) * self.sigma);
```

Comment:
Could you change the message to mention "tail corrections"? Something like `WCA cutoff ({}) is smaller than the tail corrections cutoff ({})`?